api/routers/v1/timelog.py

```diff
@@ -66,10 +67,52 @@ async def add_template(
     else:
         if template.user_id and template.user_id != current_user.id:
             raise HTTPException(status_code=403, detail="You are not authorized to create templates for this user")
+    if template.task_type:
+        task_type_valid = TaskTypeService(db).slug_is_valid(template.task_type)
+        if not task_type_valid:
+            raise HTTPException(status_code=422, detail=f"Task type {template.task_type} does not exist.")
     result = TemplateService(db).create_template(template)
     return result
 
 
+@router.put("/templates", response_model=TemplateSchema)
```

Comment:
For this case I think it would be nice to have the template_id in the url:

```suggestion
@router.put("/templates/{template_id}", response_model=TemplateSchema)
```